java-checks/src/main/java/org/sonar/java/symexecengine/ExecutionState.java

```diff
@@ -63,9 +65,15 @@ public void defineSymbol(Symbol symbol) {
   }
 
   public ExecutionState merge(ExecutionState executionState) {
-    for (Symbol symbol : executionState.reachableValues.keys()) {
+    for (Symbol symbol : executionState.allReachableValues.keys()) {
+      // FIXME(merciesa) this copies the accessible values from the parent state and is memory inefficient.
```

Comment:
![New](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/new.png) ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)